src/lambdaisland/embedkit.clj

```diff
@@ -163,7 +203,7 @@
                       titled? false
                       filters {}}
                  :as opts}]
-  (when-let [key (:secret-key conn)]
+  (when-let [key (secret-key conn)]
```

Comment:
refactor to use the protocol methods, we can no longer assume that conn is a map/record, all access has to go through the interface.